NEWS.md

```diff
@@ -1,5 +1,8 @@
 # dbplyr (development version)
 
+* `str_sub()` and `substring()` get better translations when start is negative,
+  and work on MS SQL when `end = -1` (#577).
```

Comment:
It might be good to change the title of #577 so folks don't get the mistaken idea that these functions now support dynamic start and stop